# Test Ridge Regression:

#### 1. Use the cpu.csv dataset

In [1]:
from si.io.csv_file import read_csv

In [2]:
df_path = r"C:\Users\Mariana\si\datasets\cpu\cpu.csv"
df = read_csv(df_path, sep = ',', features = True, label = True) 

#### 2. Divide the dataset into train and test sets

In [3]:
from si.model_selection.split import train_test_split

In [4]:
train_dataset, test_dataset = train_test_split(df, test_size=0.33, random_state=42)

In [5]:
print(f'(As dimensões dos dados de treino correspondem a:', train_dataset.shape())
print(f'(As dimensões dos dados de teste correspondem a:',test_dataset.shape())

(As dimensões dos dados de treino correspondem a: (141, 6)
(As dimensões dos dados de teste correspondem a: (68, 6)


Os dados de treino apresentam 141 linhas e 6 colunas, enquanto que os dados de teste apresentam 68 linhas e 6 colunas.

#### 3. Train the model. Which score do you get? And the cost?

In [6]:
from si.models.ridge_regression import RidgeRegression

In [7]:
model = RidgeRegression()

In [8]:
model.fit(train_dataset)

In [9]:
print(model.score(test_dataset))

18849.968330282394


In [10]:
print(model.cost(test_dataset))

9446.185821835123
